### PR TITLE
bootkube: move cloudprovider creds to openshift-config

### PR DESCRIPTION
**Pull request by deads2k**
We put irreplaceable config in the openshift-config namespace.
This gives a single place for user-maintained information include secrets from the installer.
This creates the resources, but consumers need to update themselves.

/assign @abhinavdahiya 
@derekwaynecarr

### CODE REVIEW

File: pkg/asset/templates/content/openshift/deprecated-cloud-creds-secret.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/templates/content"
+)
+
+const (
+	deprecatedCloudCredsSecretFileName = "cloud-creds-secret.yaml.template"
```
Comment (staebler) [inline pkg/asset/templates/content/openshift/deprecated-cloud-creds-secret.go]: This filename should have a prefix of "deprecated".

File: pkg/asset/templates/content/openshift/deprecated-role-cloud-creds-secret-reader.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/openshift/installer/pkg/asset"
+	"github.com/openshift/installer/pkg/asset/templates/content"
+)
+
+const (
+	deprecatedRoleCloudCredsSecretReaderFileName = "role-cloud-creds-secret-reader.yaml.template"
```
Comment (staebler) [inline pkg/asset/templates/content/openshift/deprecated-role-cloud-creds-secret-reader.go]: This filename should have a prefix of "deprecated".

**Comment by derekwaynecarr** [comment]
I agree with this change, we should also think about the following secrets in kube-system: 
`kube-admin` -> `openshift-config`
`kube-cloud-config` -> ? (i suspect this is the cloud-conf file, which should be derived from infra object in future)

**Comment by abhinavdahiya** [comment]
@derekwaynecarr @deads2k is this required for 4.0 ?

**Pull request closed without merging**